### DISCUSSION TITLE
C# Dialogue handler

### DISCUSSION THREAD

**Discussion by waqtastic** [community/community#21369] [Other Feature Feedback, Questions, & Ideas]
<p>Hello,</p>
<p>Just to preface this I’m really new to programming - </p>
<p>So I’m trying to create a text adventure using the console in C#.</p>
<p>I’ve made a dialogue handler that changes the colour using <code>if else</code> and <code>if</code> statements:</p>
<pre><code>public static void Dialogue(string message, string color)

		{
			if (color == "red")
			{ Console.ForegroundColor = ConsoleColor.Red; }
			else if (color == "green")
			{ Console.ForegroundColor = ConsoleColor.Green; }
			else if (color == "yellow")
			{ Console.ForegroundColor = ConsoleColor.Yellow; }
			else if (color == "cyan")
			{ Console.ForegroundColor = ConsoleColor.Cyan; }
			else if (color == "blue")
			{ Console.ForegroundColor = ConsoleColor.Blue; }
			else
			{ Console.ForegroundColor = ConsoleColor.White; }

			Console.WriteLine(message);
			Console.ResetColor();
		}
</code></pre>
<p>But want it to have expanded capability a little. I’m just not sure how to go about this - I’d like after each dialogue event that it did Console.ReadKey(); as part of it.</p>
<p>Also, but if a side question - is there a way to use two functions at the same time for a single piece of text? I wanted to add a typewriter style effect to text so it doesn’t just pop up but I’m not sure how to go about that.</p>
<p>Thanks in advance, any help is greatly appreciated</p>

**Comment by fire-eggs** [comment, marked as answer]
<p>&gt;Also, but if a side question - is there a way to use two functions at the</p>
<p>&gt;same time for a single piece of text?</p>
<p>You can call as many functions for a piece of text as you want:</p>
<pre><code>string msg = "Here is some text";
typewriter(msg);
Dialogue(msg, "cyan");
</code></pre>
<p>&gt;I wanted to add a typewriter style effect to text so it doesn’t just pop up</p>
<p>&gt;but I’m not sure how to go about that.</p>
<p>I googled “typewriter effect C#” and found <a href="https://social.msdn.microsoft.com/Forums/vstudio/en-US/518bd41d-f344-4e00-b530-9fbea5c0b867/quottypewriterquot-like-effect-in-a-c-console-application?forum=csharpgeneral" rel="nofollow noopener">a forum thread</a> which shows how you might create a typewriter-like effect.  Or <a href="http://programmingisfun.com/typewriter-effect/" rel="nofollow noopener">this one</a>. Except both of them are missing a  final “Console.WriteLine();” call to move to the next line.</p>
<p>You could add a Console.ReadKey(); call at the end of your Dialogue() function to pause for user response.</p>

**Comment by waqtastic** [comment]
<p>Thanks for this, this is definitely what I needed :slight_smile:</p>
